test/clojure/nrepl/core_test.clj

```diff
@@ -299,7 +299,8 @@
   (let [{:keys [out status value]} (combine-responses (repl-eval session "(missing paren"))]
     (is (nil? value))
     (is (= #{"done" "eval-error"} status))
-    (is (re-seq #"EOF while reading" (first (repl-values session "(.getMessage *e)"))))))
+    (is (re-seq #"EOF while reading" (first (repl-values session "(or (.getMessage *e)
+                                                                      (-> *e Throwable->map :cause))"))))))
```

Comment:
One more thing - do we need this `.getMessage` at all? I noticed that `Throwable->map` exists in Clojure 1.7 and format of the exception seems the same to me there.